pkg/publish/kind/write.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kind
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+
+	"sigs.k8s.io/kind/pkg/cluster"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+)
+
+// Tag adds a tag to an already existent image.
+func Tag(src, dest name.Tag) error {
+	return onEachNode(func(n nodes.Node) error {
+		cmd := n.Command("ctr", "--namespace=k8s.io", "images", "tag", "--force", src.String(), dest.String())
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to tag image: %w", err)
+		}
+		return nil
+	})
+}
+
+// Write saves the image into the kind nodes as the given tag.
+func Write(tag name.Tag, img v1.Image) error {
+	return onEachNode(func(n nodes.Node) error {
+		pr, pw := io.Pipe()
+		go func() {
+			pw.CloseWithError(tarball.Write(tag, img, pw))
+		}()
+
+		cmd := n.Command("ctr", "--namespace=k8s.io", "images", "import", "-").SetStdin(pr)
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to load image: %w", err)
+		}
+		return nil
+	})
+}
+
+// onEachNode executes the given function on each node. Exits on first error.
+func onEachNode(f func(nodes.Node) error) error {
+	nodeList, err := getNodes()
+	if err != nil {
+		return err
+	}
+
+	for _, n := range nodeList {
+		if err := f(n); err != nil {
+			return err
```

Comment:
Maybe include a node identifier in this error message? Or does the error include details about the node that encountered the error?